frontend/test/resolution/testModuleInitOrder.cpp

```diff
@@ -179,6 +321,145 @@ static void test1(void) {
   std::cout << std::endl;
 }
 
+static void testImportSub(void) {
+  Context::Configuration config;
+  config.chplHome = chplHome();
+  Context context(config);
+  Context* ctx = &context;
+  ErrorGuard guard(ctx);
+
+  // Do NOT set the module search paths for this test. Do not want to
+  // compute the standard/internal modules in this ordering.
+  /** setupModuleSearchPaths(ctx, false, false, {}, {}); */
+
+  auto path = TEST_NAME(ctx);
+  std::cout << path.c_str() << std::endl;
+
+  // test that dead modules / submodules do not contribute
+  // to the initialization ordering.
+  std::string contents =
+    R""""(
+    module Main {
+      import this.Library.Submodule;
+      proc main() { }
+      module Library {
+        module Submodule { }
+      }
+    }
+    )"""";
+
+  setFileText(ctx, path, contents);
+
+  // Get the module.
+  auto& br = parseAndReportErrors(ctx, path);
+  assert(br.numTopLevelExpressions() == 1);
+  auto m1 = br.topLevelExpression(0)->toModule();
+  assert(m1);
+
+  // check to make sure the initialization order is correct.
+  checkModuleInitOrder(ctx, m1->id(), "Library", "Submodule", "Main");
+
+  std::cout << "---" << std::endl;
+
+  assert(!guard.realizeErrors());
+  std::cout << std::endl;
+}
+
+static void testMentionedSubmodule(void) {
+  Context::Configuration config;
+  config.chplHome = chplHome();
+  Context context(config);
+  Context* ctx = &context;
+  ErrorGuard guard(ctx);
+
+  // Do NOT set the module search paths for this test. Do not want to
+  // compute the standard/internal modules in this ordering.
+  /** setupModuleSearchPaths(ctx, false, false, {}, {}); */
+
+  auto path = TEST_NAME(ctx);
+  std::cout << path.c_str() << std::endl;
+
+  // test that dead modules / submodules do not contribute
+  // to the initialization ordering.
+  std::string contents =
+    R""""(
+    module Main {
+      module Submodule {
+        proc foo() { }
+      }
+
+      proc main() {
+        Submodule.foo();
+      }
```

Comment:
What if `Main` has two submodules, with `Sub2` having a `main` proc that refers to `Sub1.foo`?

Reply:
You can't write that without a use/import statement because parent module symbols aren't visible unless you use/import the parent.